lib/autoproj/daemon/pull_request_cache.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'autoproj'
+require 'autoproj/daemon/github/pull_request'
+require 'yaml'
+
+module Autoproj
+    # Main plugin module
+    module Daemon
+        # A PullRequest cache
+        class PullRequestCache
+            # @return [Array<CachedPullRequest>]
+            attr_reader :pull_requests
+
+            # @return [Autoproj::Workspace]
+            attr_reader :ws
+
+            def initialize(workspace)
+                @pull_requests = []
+                @ws = workspace
+            end
+
+            CachedPullRequest = Struct.new :base_owner, :base_name,
+                                           :number, :base_branch, :head_sha, :overrides
+
+            # @param [CachedPullRequest] cache
+            # @param [Github::PullRequest] pull_request
+            # @return [Boolean]
+            def same?(cache, pull_request)
+                cache.base_owner == pull_request.base_owner &&
+                    cache.base_name == pull_request.base_name &&
+                    cache.number == pull_request.number
+            end
+
+            # @param [Github::PullRequest] pull_request
+            # @return [nil]
+            def add(pull_request, overrides)
+                pull_requests.delete_if { |pr| same?(pr, pull_request) }
+                pull_requests << CachedPullRequest.new(
+                    pull_request.base_owner,
+                    pull_request.base_name,
+                    pull_request.number,
+                    pull_request.base_branch,
+                    pull_request.head_sha,
+                    overrides
+                )
+            end
+
+            # @return [Array<CachedPullRequest>]
```

Comment:
It doesn't return anything. This is not typescript.

If you want to document "don't use the return value", YARD has `@return [void]`

Reply:
> If you want to document "don't use the return value", YARD has `@return [void]`

That's what I want :+1: